pentestportal_sg.tf

```diff
@@ -62,7 +62,7 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_pentestportal_via_a
   # for_each will only accept a map or a list of strings, so we have
   # to do a little finagling to get the list of port objects into an
   # acceptable form.
-  for_each = { for index, d in var.inbound_ports_allowed["pentestportal"] : index => d }
+  for_each = { for d in var.inbound_ports_allowed["pentestportal"] : format("%s_%d_%d", d.protocol, d.from_port, d.to_port) => d }
```

Comment:
Just a note but I would really like to see more specific variable typing for this variable (`inbound_ports_allowed`) since we are relying on specific attributes.